src/dotnet-interactive-vscode/common/interactiveClient.ts

```diff
@@ -316,8 +320,9 @@ export class InteractiveClient {
         return new Promise<TEvent>(async (resolve, reject) => {
             let handled = false;
             token = token || this.getNextToken();
+            const id = Guid.create().toString();
             let disposable = this.subscribeToKernelTokenEvents(token, eventEnvelope => {
-                if (eventEnvelope.command?.token === token) {
+                if (eventEnvelope.command?.id === id) {
```

Comment:
Why this change and are there tests that demonstrate that this is correct?

Reply:
We have tests covering that part of the code. The change is because we want to terminate only when we obtain termination events from the command vscode extension generated via the client. We also tested using the notebook we have to exercise the feature set.

Reply:
needs better testing